src/main/scala/org/constellation/snapshotstreaming/FileBasedLastGlobalSnapshotStorage.scala

```diff
@@ -0,0 +1,82 @@
+package org.constellation.snapshotstreaming
+
+import java.nio.file.NoSuchFileException
+import cats.effect.Async
+import cats.effect.std.Semaphore
+import cats.syntax.applicativeError._
+import cats.syntax.flatMap._
+import cats.syntax.functor._
+import cats.{Applicative, MonadThrow}
+import org.tessellation.sdk.domain.snapshot.storage.LastGlobalSnapshotStorage
+import org.tessellation.sdk.domain.snapshot.Validator.isNextSnapshot
+import org.tessellation.dag.snapshot.{GlobalSnapshot, GlobalSnapshotReference, SnapshotOrdinal}
+import org.tessellation.schema.height.Height
+import org.tessellation.security.Hashed
+import fs2.io.file._
+import fs2.{Stream, text}
+import io.circe.Codec
+import io.circe.generic.semiauto.deriveCodec
+import io.circe.parser.decode
+import io.circe.syntax._
+import org.typelevel.log4cats.slf4j.Slf4jLogger
+
+object FileBasedLastGlobalSnapshotStorage {
+
+  def make[F[_]: Async: Files](path: Path): F[LastGlobalSnapshotStorage[F]] =
+    Semaphore[F](1)
+      .map(make(path, _))
+
+  private def make[F[_]: Async: Files](path: Path, semaphore: Semaphore[F]): LastGlobalSnapshotStorage[F] =
+    new LastGlobalSnapshotStorage[F] {
+
+      implicit val codec: Codec[Hashed[GlobalSnapshot]] = deriveCodec[Hashed[GlobalSnapshot]]
+
+      private val logger = Slf4jLogger.getLogger[F]
+
+      def set(snapshot: Hashed[GlobalSnapshot]): F[Unit] =
+        semaphore.permit.use { _ =>
+          get.flatMap {
+            case Some(last) if isNextSnapshot(last, snapshot) =>
+              Stream
+                .emit(snapshot.asJson.spaces2)
```

Comment:
why we need `spaces2` if we're not going to read this file manually?

Reply:
We may look into it in some cases and it doesn't break anything.

Reply:
we can use `jq` then

Reply:
We can. I can address that if you want. Personally I don't see that as a problem or being worse/better.